lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -1985,6 +2000,41 @@ void DxilGenerationPass::AddCreateHandleForPhiNode(std::unordered_map<Instructio
       phi->setIncomingValue(i, undefObj);
     }
   }
+
+  // Create sel for each operands of the createHandle.
+  // Generate createHandle with sel operand.
+  // Map createHandle to original sel.
+  for (SelectInst *Sel : objSelectList) {
+    Value *Cond = Sel->getCondition();
+    Value * ValT= Sel->getTrueValue();
+    if (!isa<Instruction>(ValT) || !handleMap.count(cast<Instruction>(ValT))) {
```

Comment:
There is a lot of repeated code patterns in this part of the code. Would be good to try and remove the redundancy.